source/common/memory/stats.cc

```diff
@@ -107,6 +107,20 @@ void Stats::dumpStatsToLog() {
 #endif
 }
 
+std::string Stats::dumpStats() {
+#if defined(TCMALLOC)
+  return tcmalloc::MallocExtension::GetStats();
+#elif defined(GPERFTOOLS_TCMALLOC)
+  constexpr int buffer_size = 100000;
+  std::string buffer(buffer_size, '\0');
```

Comment:
I think make_unique was used in the other place because of concerns of stack overflow?